Doctrine/Orm/Filter/OrderFilter.php

```diff
@@ -68,19 +70,39 @@ public function apply(ResourceInterface $resource, QueryBuilder $queryBuilder)
         }
 
         $properties = $this->extractProperties($request);
-        $fieldNames = array_flip($this->getClassMetadata($resource)->getFieldNames());
 
         foreach ($properties as $property => $order) {
-            if (!$this->isPropertyEnabled($property) || !isset($fieldNames[$property])) {
+            if (!$this->isPropertyEnabled($property) || !$this->isPropertyMapped($property, $resource)) {
                 continue;
-            } elseif ('' === $order && isset($this->properties[$property])) {
+            }
+
+            if ('' === $order && isset($this->properties[$property])) {
                 $order = $this->properties[$property];
             }
 
             $order = strtoupper($order);
-            if ('ASC' === $order || 'DESC' === $order) {
-                $queryBuilder->addOrderBy(sprintf('o.%s', $property), $order);
+            if (!in_array($order, ['ASC', 'DESC'])) {
+                continue;
```

Comment:
This has always been the behavior (to silently ignore). The built-in filters are pretty consistent on this (I haven't looked through thoroughly to see if there are cases where it might just throw an exception or error out).